app/models/package_manager/maven/google.rb

```diff
@@ -100,16 +106,30 @@ def self.update_all_versions
 
         pp "added project #{name}"
 
-        versions.each do |version|
-          version_hash = one_version_for_name(version, name)
-
-          add_version(db_project, version_hash)
-
-          pp "added version #{version}"
+        api_versions = version_numbers
+          .map { |version_number| one_version_for_name(version_number, name) }
+          .map { |mapped_version| version_hash_to_version_object(mapped_version) }
+
+        retried = false
+        begin
+          BulkVersionUpdater.new(
+            db_project: db_project,
+            api_versions: api_versions,
+            repository_source_name: self::HAS_MULTIPLE_REPO_SOURCES ? [self::REPOSITORY_SOURCE_NAME] : nil
+          ).run!
         rescue Faraday::ConnectionFailed
-          retry
+          retry unless retried
+          StructuredLog.capture(
+            "GOOGLE_MAVEN_VERSION_UPSERT_FAILURE",
+            {
+              project_id: db_project.id,
+              versions: api_versions.map(&:version_number),
+            }
+          )
         end
 
+        pp "added versions #{api_versions.map(&:version_number).join(', ')}"
```

Comment:
`pp`

Reply:
before I remove this, do you remember if there was a reason for it? https://github.com/librariesio/libraries.io/commit/349b6fb896aa45fdd5ffc794077eaab6d495185e#diff-b3d7088ca715b858cc9b988967128f4f0fb6a117c0633b2e0bc7fb4926bda26fR108

Reply:
Oh, right, this was for debugging since this is run from a Rake task. This can stay.